packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -26,15 +26,18 @@ use crate::{
 };
 
 /// A Postgres-backed store
-pub struct PostgresStore {
-    connection: PoolConnection<Postgres>,
+pub struct PostgresStore<C: AsClient> {
+    client: C,
 }
 
-impl PostgresStore {
+impl<C> PostgresStore<C>
+where
+    C: AsClient,
```

Comment:
We can't use `GenericClient` as the return value of the pool does not implement it. `AsClient` is manually implemented for all.